### PR TITLE
Improving calculation for total job memory

### PR DESCRIPTION
**Pull request by opotowsky**
## What is the change?

This swaps the calculation for total job memory to use a more straightforward cluster resource request parameter. Some edits and additions to the tests flow from the update.

## Why is the change being made?

This makes the math simpler!

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] ~The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.~ too small to matter
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
Wait, neither of these settings exist in ARMI:

```python
self.cs["nTasksPerNode"]  # OLD
self.cs["cpusPerTask"]  # NEW
```

If we are modifying this code, I would prefer to move it out of ARMI.

**Review by john-science — APPROVED** [review]

**Review by albeanth — COMMENTED** [review]
Can you update the docstring for test_getTotalJobMemory?

File: armi/bookkeeping/tests/test_memoryProfiler.py [hunk 1]
```diff
@@ -166,20 +175,26 @@ def test_printCurrentMemoryState(
...
-            csMock = MagicMock()
-            csMock.__getitem__.return_value = 2
-            self.memPro.cs = csMock
+            self.memPro.cs = {"cpusPerTask": 1, "nTasks": 10}
             self.memPro.printCurrentMemoryState()
             stdOut = mockLogs.getStdout()
             self.assertIn("Currently using 6.0 GB of memory.", stdOut)
             self.assertIn("There is 44.0 GB of memory left.", stdOut)
             self.assertIn("There is a total allocation of 50.0 GB", stdOut)
+            # Try another for funzies
```
Comment (albeanth) [inline armi/bookkeeping/tests/test_memoryProfiler.py:184]:
```suggestion
            # Try another for funzies where we only use half the available resources on the node
```

**Comment by opotowsky** [comment]
> Can you update the docstring for test_getTotalJobMemory?

I'm not sure how it's inaccurate?

**Comment by opotowsky** [comment]
> Wait, neither of these settings exist in ARMI:
> 
> ```python
> self.cs["nTasksPerNode"]  # OLD
> self.cs["cpusPerTask"]  # NEW
> ```
> 
> If we are modifying this code, I would prefer to move it out of ARMI.

Discussed offline. Removal will happen! Just not in this PR.

**Review by albeanth — APPROVED** [review]

**Comment by albeanth** [comment]
> > Can you update the docstring for test_getTotalJobMemory?
> 
> I'm not sure how it's inaccurate? It's the same test as before, just with the new inputs. (and I added a combo or two)

you're right! I was wrong. All good!

**Pull request merged**